### ISSUE TITLE
Faster_R-CNN_R50_FPN Training on Custom Dataset

### ISSUE THREAD

**Issue by axlecky** [facebookresearch/detectron2#4230]
I am trying to train a Faster R-CNN R50 through Detectron2 Pytorch on Google Colab. I have a dataset consisting of 5000 training images, 1400 validation images, and 700 test images. Can I know how many iterations should I set? And also what learning rate should I set? Thank you.

Code:
`from detectron2.config import get_cfg
#from detectron2.evaluation.coco_evaluation import COCOEvaluator
import os

cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-Detection/faster_rcnn_R_50_FPN_3x.yaml"))
cfg.DATASETS.TRAIN = ("my_dataset_train",)
cfg.DATASETS.TEST = ("my_dataset_val",)

cfg.DATALOADER.NUM_WORKERS = 4
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-Detection/faster_rcnn_R_50_FPN_3x.yaml")  # Let training initialize from model zoo
cfg.SOLVER.IMS_PER_BATCH = 4
cfg.SOLVER.BASE_LR = 0.001

cfg.SOLVER.WARMUP_ITERS = 1000
cfg.SOLVER.MAX_ITER = 1500 #adjust up if val mAP is still rising, adjust down if overfit
cfg.SOLVER.STEPS = (1000, 1500)
cfg.SOLVER.GAMMA = 0.05

cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 64
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 17 #your number of classes + 1

cfg.TEST.EVAL_PERIOD = 500

os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
trainer = CocoTrainer(cfg)
trainer.resume_or_load(resume=False)
trainer.train()`

**Comment by StudioTV** [comment]
Hey !

It always depends on your dataset, there is no "perfect parameters" that works for every dataset sadly. You have to try different settings and see which one produces the best results !
But I can give you one tip, plot your loss for training and validation, then you can track when your model is overfitting, it will give you an idea of the number of iterations.

`cfg.MODEL.ROI_HEADS.NUM_CLASSES = 17 #your number of classes + 1`

Are you sure about the +1, you can read the official documentation here  : https://detectron2.readthedocs.io/en/latest/tutorials/datasets.html
